src/pipx/commands/install.py

```diff
@@ -185,42 +186,52 @@ def install_all(
     force: bool,
 ) -> ExitCode:
     """Return pipx exit code."""
+    failed: List[str] = []
     venv_container = VenvContainer(paths.ctx.venvs)
 
     for venv_metadata in extract_venv_metadata(spec_metadata_file):
         # Install the main package
         main_package = venv_metadata.main_package
         venv_dir = venv_container.get_venv_dir(f"{main_package.package}{main_package.suffix}")
-        install(
-            venv_dir,
-            None,
-            [generate_package_spec(main_package)],
-            local_bin_dir,
-            local_man_dir,
-            python or get_python_interpreter(venv_metadata.source_interpreter),
-            pip_args,
-            venv_args,
-            verbose,
-            force=force,
-            reinstall=False,
-            include_dependencies=main_package.include_dependencies,
-            preinstall_packages=[],
-            suffix=main_package.suffix,
-        )
-
-        # Install the injected packages
-        for inject_package in venv_metadata.injected_packages.values():
-            commands.inject(
+        try:
+            package_exit = 0
+            package_exit |= install(
                 venv_dir,
                 None,
-                [generate_package_spec(inject_package)],
+                [generate_package_spec(main_package)],
+                local_bin_dir,
+                local_man_dir,
+                python or get_python_interpreter(venv_metadata.source_interpreter),
                 pip_args,
-                verbose=verbose,
-                include_apps=inject_package.include_apps,
-                include_dependencies=inject_package.include_dependencies,
+                venv_args,
+                verbose,
                 force=force,
-                suffix=inject_package.suffix == main_package.suffix,
+                reinstall=False,
+                include_dependencies=main_package.include_dependencies,
+                preinstall_packages=[],
+                suffix=main_package.suffix,
             )
 
+            # Install the injected packages
+            for inject_package in venv_metadata.injected_packages.values():
+                package_exit |= commands.inject(
+                    venv_dir,
+                    None,
+                    [generate_package_spec(inject_package)],
+                    pip_args,
+                    verbose=verbose,
+                    include_apps=inject_package.include_apps,
+                    include_dependencies=inject_package.include_dependencies,
+                    force=force,
+                    suffix=inject_package.suffix == main_package.suffix,
+                )
+        except PipxError as e:
+            print(e, file=sys.stderr)
+            failed.append(venv_dir.name)
+        else:
```

Comment:
I don't think this is necessary here according to this:

https://github.com/pypa/pipx/blob/44023628756588ca74c9a2256037e796cd02613a/src/pipx/commands/install.py#L120

Reply:
For `reinstall_all`, it's a little bit different than this one cause it needs to set `first_reinstall` to False: https://github.com/pypa/pipx/blob/main/src%2Fpipx%2Fcommands%2Freinstall.py#L136-L136

Reply:
OK, will do.

Reply:
Tracked in #1357